h2o-r/tests/testdir_algos/stackedensemble/runit_stackedensemble_custom_metalearner_params.R

```diff
@@ -0,0 +1,119 @@
+    print("SE with GBM metalearner and custom params")
+    stack_gbm <- h2o.stackedEnsemble(x = x,
+    y = y,
+    training_frame = train,
+    base_models = list(my_gbm, my_rf),
+    metalearner_algorithm = "gbm",
+    metalearner_params = gbm_params)
+    # Check that metalearner_algorithm is a default GBM
+    expect_equal(stack_gbm@parameters$metalearner_algorithm, "gbm")
+    expect_equal(stack_gbm@allparameters$metalearner_algorithm, "gbm")
+    # Check that the metalearner is default GBM
+    meta_gbm <- h2o.getModel(stack_gbm@model$metalearner$name)
+    expect_equal(meta_gbm@algorithm, "gbm")
+    expect_equal(meta_gbm@parameters$ntrees, 100)
+    expect_equal(meta_gbm@parameters$max_depth, 6)
+
+
+    # Train a stacked ensemble & metalearner_algorithm "drf"; check that metalearner_algorithm works with CV
+    print("SE with DRF metalearner and custom params")
+    stack_drf <- h2o.stackedEnsemble(x = x,
+    y = y,
+    training_frame = train,
+    base_models = list(my_gbm, my_rf),
+    metalearner_nfolds = 3,
+    metalearner_algorithm = "drf",
+    metalearner_params = drf_params)
+    # Check that metalearner_algorithm is a default RF
+    expect_equal(stack_drf@parameters$metalearner_algorithm, "drf")
+    # Check that CV was performed
+    expect_equal(stack_drf@allparameters$metalearner_nfolds, 3)
+    meta_drf <- h2o.getModel(stack_drf@model$metalearner$name)
+    expect_equal(meta_drf@algorithm, "drf")
+    expect_equal(meta_drf@allparameters$nfolds, 3)
+    expect_equal(meta_drf@parameters$ntrees, 100)
+    expect_equal(meta_drf@parameters$max_depth, 6)
+
+
+    # Train a stacked ensemble & metalearner_algorithm "glm"
+    print("SE with GLM metalearner and custom params")
+    stack_glm <- h2o.stackedEnsemble(x = x,
+    y = y,
+    training_frame = train,
+    base_models = list(my_gbm, my_rf),
+    metalearner_algorithm = "glm",
+    metalearner_params = glm_params)
+    # Check that metalearner_algorithm is a default GLM
+    expect_equal(stack_glm@parameters$metalearner_algorithm, "glm")
+    meta_glm <- h2o.getModel(stack_glm@model$metalearner$name)
+    expect_equal(meta_glm@algorithm, "glm")
+    expect_equal(meta_glm@parameters$alpha, 0)
+    expect_equal(meta_glm@parameters$lambda, 0)
+
+
+    # Train a stacked ensemble & metalearner_algorithm "deeplearning"
+    print("SE with DL metalearner and custom params")
+    stack_deeplearning <- h2o.stackedEnsemble(x = x,
+    y = y,
+    training_frame = train,
+    base_models = list(my_gbm, my_rf),
+    metalearner_algorithm = "deeplearning",
+    metalearner_params = dl_params)
+    # Check that metalearner_algorithm is a default DNN
+    expect_equal(stack_deeplearning@parameters$metalearner_algorithm, "deeplearning")
+    meta_dl <- h2o.getModel(stack_deeplearning@model$metalearner$name)
+    expect_equal(meta_dl@algorithm, "deeplearning")
+    expect_equal(meta_dl@parameters$hidden, c(32,32,32))
+    expect_equal(meta_dl@parameters$epochs, 20)
+
+}
```

Comment:
Same comment from the pyunit above about adding some tests that try to break this and see if we handle it gracefully.  Or if you manually test it and see that we don't handle this gracefully, you could change our error-handling on this and then just skip the tests once it's fixed.

Reply:
Yep, can test it out locally first to see what happens before a test is made.

Reply:
Basically it will be ignored if set to AUTO and you will hit an illegal argument error if you pass in incorrect params.

Reply:
👍